src/_pytest/config/__init__.py

```diff
@@ -1029,8 +1045,8 @@ def _initini(self, args: Sequence[str]) -> None:
             rootdir_cmd_arg=ns.rootdir or None,
             config=self,
         )
-        self.rootdir = py.path.local(str(rootpath))
-        self.inifile = py.path.local(str(inipath)) if inipath else None
+        self.rootpath = rootpath
```

Comment:
While at it, we might decide to also make those read-only properties and hide them behind private attributes `_rootpath` and `_inipath`.

Reply:
I didn't do this one, it feels a bit too heavy handed/boilerplate-y to me ATM. I did think to add a `Final` type annotation but it doesn't work in this case because these attributes are not assigned in the `__init__`, only later.

Reply:
> I didn't do this one, it feels a bit too heavy handed/boilerplate-y to me ATM

TBH I prefer we do this. We don't want users to go around and change those attributes at all, I would rather to get this *right* from the start.

Reply:
OK, changed to properties.

Reply:
Thanks a lot! 🙇